### PR TITLE
[22.05] Pass ``--cleanenv`` by default to singularity command

### LINKED ISSUE #14408: Singularity copies most host environment variables into container
**Describe the bug**
Singularity copies most host environment variables into the container.
I am unsure if this had always been the case, but it doesn't seem like a good idea.
There is the `--cleanenv` option that prevents this behavior, we should use it.
That is closer to how docker works.

**Galaxy Version and/or server at which you observed the bug**
22.05, but should affect all releases

**To Reproduce**
1. Run a tool that prints the environment, and see that it also includes host variables

**Expected behavior**
Only explicitly set environment variables should be passed into the container

**Additional context**
Might be part of the problem that caused https://github.com/galaxyproject/tools-iuc/pull/4698

### PR DESCRIPTION
**Pull request by mvdbeek**
This is more in line with docker and fixes #14408 

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:
Set up singularity and see that the `--cleanenv` flag is being passed to singularity.

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/config/sample/job_conf.xml.sample_advanced [hunk 1]
```diff
@@ -606,6 +606,11 @@
           <!-- Following option can be used to tweak sudo command used by
                default. -->
           <!-- <param id="singularity_sudo_cmd">/usr/bin/sudo -extra_param</param> -->
+          <!-- Singularity by default passes most host environment variables into the container.
+               This may be a security or configuration issue, hence galaxy passes the `cleanenv`
+               argument by default. You can turn this off by setting singularity_cleanenv to `false`.
+          -->
+          <!-- <param id="singulairty_cleanenv">true</param> -->
```
Comment (nsoranzo) [inline lib/galaxy/config/sample/job_conf.xml.sample_advanced]:
```suggestion
          <!-- <param id="singularity_cleanenv">true</param> -->
```

Comment (mvdbeek) [reply lib/galaxy/config/sample/job_conf.xml.sample_advanced]: Thanks!

**Review by bernt-matthias — APPROVED** [review]
Is there a way to test this (that the parameter is really set and that env variables from outside of the container don't affect the container env)

**Review by dannon — APPROVED** [review]

**Comment by mvdbeek** [comment]
Yes, but for that we'd have to find a decent way of installing singularity on the action runners. We should do that in general, back on jenkins we did actually run the singularity tests that we have.

**Comment by bernt-matthias** [comment]
Hi @mvdbeek .. I'm wondering if with this change environment variables that are set in the destination are disabled (i.e. not available in the container). I was just looking at a job script and it appeared that such env variables are not set in singularity execution...

**Comment by mvdbeek** [comment]
Destination vars should be set here:
https://github.com/galaxyproject/galaxy/blob/8865afc44dacf711026e9acae8403b735aea85ff/lib/galaxy/tool_util/deps/singularity_util.py#L69-L72
`--cleanenv` is an additional option.

**Comment by bernt-matthias** [comment]
In my case the destination has:

```
            <env id="ALPHAFOLD_DB">/data/db/alphafold/2.1.2/2022-11-14/</env>
            <env id="ALPHAFOLD_USE_GPU">False</env>
```

I do see the lines setting those in `galaxy_XYZ.sh`

```
ALPHAFOLD_DB="/data/db/alphafold/2.1.2/2022-11-14/"; export ALPHAFOLD_DB
ALPHAFOLD_USE_GPU="False"; export ALPHAFOLD_USE_GPU
```

But the singularity call in `galaxy_XYZ.sh` is only (ie misses the `ALPHAFOLD` vars):

```
SINGULARITYENV_GALAXY_SLOTS=$GALAXY_SLOTS SINGULARITYENV_GALAXY_MEMORY_MB=$GALAXY_MEMORY_MB SINGULARITYENV_GALAXY_MEMORY_MB_PER_SLOT=$GALAXY_MEMORY_MB_PER_SLOT SINGULARITYENV_HOME=$HOME SINGULARITYENV__GALAXY_JOB_HOME_DIR=$_GALAXY_JOB_HOME_DIR SINGULARITYENV__GALAXY_JOB_TMP_DIR=$_GALAXY_JOB_TMP_DIR SINGULARITYENV_TMPDIR=$TMPDIR SINGULARITYENV_TMP=$TMP SINGULARITYENV_TEMP=$TEMP singularity -s exec ...
```

This should break if `--cleanenv` is used.

**Comment by mvdbeek** [comment]
Right, I think that's quite intentional, you'll have to define these as `SINGULARITYENV_ALPHAFOLD_USE_GPU ` or as `<param id={singularity,docker}_env_ALPHAFOLD_USE_GPU>false</param>`. Or turn the option off if you're sure there's no secrets or settings you want to exclude from the container.

**Comment by bernt-matthias** [comment]
I completely agree that it might be cool that there is an (undocumented) mechanism to set additional variables that only affect destinations affecting containerized execution.

But for environment variables in general this should not be the solution IMHO .. for the following reasons: 

1. Env vars are [documented](https://docs.galaxyproject.org/en/master/admin/jobs.html#environment-modifications) to affect any destination, regardless of execution in a container or not.
2. The `{singularity,docker}_env_` seems completely undocumented

I guess I should also add that the tool in question uses an explicit container requirement .. and I just noticed that I had not set `<param id="singularity_enabled">true</param>` in the destination .. but still singularity was used :) .. odd. Adding this `param` does not change anything regarding the env variables.

**Pull request merged**